src/main/java/org/java_websocket/extensions/permessage_deflate/PerMessageDeflateExtension.java

```diff
@@ -53,23 +53,28 @@ public class PerMessageDeflateExtension extends CompressionExtension {
   // For WebSocketClients, this variable holds the extension parameters that client himself has requested.
   private Map<String, String> requestedParameters = new LinkedHashMap<>();
 
-  private Inflater inflater = new Inflater(true);
-  private Deflater deflater = new Deflater(Deflater.DEFAULT_COMPRESSION, true);
-
-  public Inflater getInflater() {
-    return inflater;
-  }
+  private int deflaterLevel = Deflater.DEFAULT_COMPRESSION;
 
-  public void setInflater(Inflater inflater) {
-    this.inflater = inflater;
-  }
+  private Inflater inflater = new Inflater(true);
+  private Deflater deflater = new Deflater(this.deflaterLevel, true);
 
-  public Deflater getDeflater() {
-    return deflater;
+  /**
+   * Get the compression level used for the compressor.
+   * @return the compression level (0-9)
+   */
+  public int getDeflaterLevel() {
+    return this.deflaterLevel;
   }
 
-  public void setDeflater(Deflater deflater) {
-    this.deflater = deflater;
+  /**
+   * Set the compression level used for the compressor.
+   * @param level the compression level (0-9)
+   */
+  public void setDeflaterLevel(int level) {
+    this.deflater.setLevel(level);
+    this.deflaterLevel = level;
+    //If the compression level is changed, the next invocation of deflate will compress the input available so far with the old level (and may be flushed); the new level will take effect only after that invocation.
```

Comment:
Unfortunately, this does not work in Java 8:

While this matches the Java 8 Deflater Javadoc, the implementation does not match the documentation:
https://github.com/openjdk/jdk8u/blob/master/jdk/src/share/classes/java/util/zip/Deflater.java

The set level is never picked up as far as I can tell. The "setParams" boolean is never evaluated. It is in Java 11, though, so it works there:
https://github.com/openjdk/jdk11u/blob/master/src/java.base/share/classes/java/util/zip/Deflater.java

So, for Java 8 compatibility, the Deflater instance has to be created with the configured compression level.

How about not having setDeflaterLevel at all, but rather making it an optional constructor argument:

private final int compressionLevel;
private final Inflater inflater;
private final Deflater deflater;

public PerMessageDeflateExtension() {
  this(Deflater.DEFAULT_COMPRESSION);
}

public PerMessageDeflateExtension(int compressionLevel) {
  this.compressionLevel = compressionLevel;
  this.inflater = new Inflater(true);
  this.deflater = new Deflater(compressionLevel, true);
}

public int getCompressionLevel() {
  return compressionLevel;
}

  @Override
  public IExtension copyInstance() {
    PerMessageDeflateExtension clone = new PerMessageDeflateExtension(this.getCompressionLevel());
     clone.setThreshold(this.getThreshold());
    clone.setClientNoContextTakeover(this.isClientNoContextTakeover());
     clone.setServerNoContextTakeover(this.isServerNoContextTakeover());
     return clone;
   }